src/storage/LockingResourceStore.ts

```diff
@@ -63,6 +63,11 @@ export class LockingResourceStore implements AtomicResourceStore {
       async(): Promise<void> => this.source.modifyResource(identifier, patch, conditions));
   }
 
+  public async resourceExists(identifier: ResourceIdentifier): Promise<boolean> {
+    return this.locks.withWriteLock(this.getLockIdentifier(identifier),
```

Comment:
This is a read operation, not a write operation so the read lock needs to be used here.